ldcpy/metrics.py

```diff
@@ -50,24 +50,20 @@ def __init__(
             lat_coord_name = ds.cf['latitude'].name
         self._lat_coord_name = lat_coord_name
 
-        # assum lat/lon (andorider:  time, level, lat, lon -
-        # where time and level may be missing)
-        if lat_dim_name is None:
-            if len(ds.dims) == 2:
-                lat_dim_name = ds.dims[0]
-            elif len(ds.dims) == 3:
-                lat_dim_name = ds.dims[1]
-            elif len(ds.dims) > 3:
-                lat_dim_name = ds.dims[3]
-        self._lat_dim_name = lat_dim_name
+        dd = ds.cf['latitude'].dims
+        ll = len(dd)
+        if ll == 1:
+            if lat_dim_name is None:
+                lat_dim_name = dd[0]
+            if lon_dim_name is None:
+                lon_dim_name = ds.cf['longitude'].dims[0]
+        elif ll == 2:
+            if lat_dim_name is None:
+                lat_dim_name = dd[0]
+            if lon_dim_name is None:
+                lon_dim_name = dd[1]
```

Comment:
Are we sure the ordering of lat and lon in dd is always consistent?

Reply:
yeah this is an annoying problem.

For POP `nlon` is `lon_dim_name` and `nlat` is `lat_dim_name` but there's no good way to discover this without setting `axis="X"` and `axis="Y"` attributes. If you did that then

``` python
lat_dim_name = ds.cf.axes["Y"][0]
lon_dim_name = ds.cf.axes["X"][0]
```

you would have to do
```
ds["nlon"] = xr.DataArray(np.arange(ds.sizes["nlon"]), dims="nlon", attrs={"axis": "X"})
```
we could add a helper function to `pop_tools` to make all these adjustments. Related discussion for the ROMS ocean model is here: https://github.com/xarray-contrib/cf-xarray/issues/84

Reply:
This should be OK, because it is not the data array dims which seemingly can be in any order, but rather the dims for the lat (and lon) coords - which for cesm are always shaped (lat, lon) as far as I know...